migrations/20240111120840-sorted-transactions.js

```diff
@@ -0,0 +1,194 @@
+          "hostCurrency",
+          SUM(
+            COALESCE("amountInHostCurrency", 0)
+            + COALESCE("platformFeeInHostCurrency", 0)
+            + COALESCE("hostFeeInHostCurrency", 0)
+            + COALESCE("paymentProcessorFeeInHostCurrency", 0)
+            + COALESCE("taxAmount" * "hostCurrencyFxRate", 0)
+          ) OVER (
+            PARTITION BY "CollectiveId", "hostCurrency"
+            ORDER BY SortDate("Transactions") ASC, "TransactionGroup" ASC, SortPriority("Transactions") ASC
+          ) as "balance"
+          FROM "Transactions", "ActiveCollectives"
+          WHERE "CollectiveId" = "ActiveCollectives"."ActiveCollectiveId"
+       )
+    `);
+
+    // Add a unique index on collective ID to the materialized view
+    await queryInterface.sequelize.query(`CREATE UNIQUE INDEX CONCURRENTLY ON "TransactionBalances"(id)`);
+
+    await recreateUnmodifiedViews(queryInterface);
+  },
+
+  async down(queryInterface) {
+    await queryInterface.sequelize.query(`DROP VIEW IF EXISTS "CurrentCollectiveBalance"`);
+
+    await queryInterface.sequelize.query(`DROP MATERIALIZED VIEW IF EXISTS "CollectiveBalanceCheckpoint"`);
+
+    await queryInterface.sequelize.query(`DROP MATERIALIZED VIEW IF EXISTS "TransactionBalances"`);
+
+    await queryInterface.sequelize.query(`DROP FUNCTION SortDate("Transactions")`);
+
+    await queryInterface.sequelize.query(`DROP FUNCTION SortPriority("Transactions")`);
+
+    // Copied from migrations/20230104225718-transaction-balances-order-by-create-date.js
+    await queryInterface.sequelize.query(`
+      CREATE MATERIALIZED VIEW "TransactionBalances" AS (
+        WITH "ActiveCollectives" AS (
+          SELECT t."CollectiveId" as "ActiveCollectiveId"
+          FROM "Transactions" t
+          LEFT JOIN  "Collectives" c ON c."id" = t."CollectiveId" AND c."deletedAt" IS NULL
+          WHERE t."deletedAt" IS NULL AND t."hostCurrency" IS NOT NULL
+          AND c."isActive" IS TRUE
+          GROUP BY t."CollectiveId"
+          HAVING COUNT(DISTINCT t."hostCurrency") = 1
+        )
+        SELECT
+          "id",
+          "CollectiveId",
+          "createdAt",
+          "hostCurrency",
+          SUM(
+            COALESCE("amountInHostCurrency", 0)
+            + COALESCE("platformFeeInHostCurrency", 0)
+            + COALESCE("hostFeeInHostCurrency", 0)
+            + COALESCE("paymentProcessorFeeInHostCurrency", 0)
+            + COALESCE("taxAmount" * "hostCurrencyFxRate", 0)
+          ) OVER (PARTITION BY "CollectiveId", "hostCurrency" ORDER BY "createdAt" ASC, "id" ASC) as "balance"
+          FROM "Transactions", "ActiveCollectives"
+          WHERE "deletedAt" IS NULL
+          AND "CollectiveId" = "ActiveCollectives"."ActiveCollectiveId"
+          ORDER BY "createdAt" ASC
+       )
+    `);
+
+    // Add a unique index on collective ID to the materialized view
+    await queryInterface.sequelize.query(`CREATE UNIQUE INDEX CONCURRENTLY ON "TransactionBalances"(id)`);
```

Comment:
The index seems to be on the transaction id, not the collective ID as per the comment

Reply:
I see this is copied code from `migrations/20230104225718-transaction-balances-order-by-create-date.js`

Later in that migration we do create an index on the collective ID, so maybe we want that index created here too

Reply:
We're carrying this bad comment for ages by copy-pasting it!

Reply:
Not clear at that point that an index on CollectiveId is useful so I'm not gonna add one here.